tests/NHSD.BuyingCatalogue.Solutions.Persistence.DatabaseTests/SupplierRepositoryTests.cs

```diff
@@ -0,0 +1,113 @@
+        private readonly string _link = "www.someLink.com";
+
+        private readonly string _newDescription = "new description";
+        private readonly string _newLink = "www.newLink.com";
+
+        [SetUp]
+        public async Task SetUp()
+        {
+            await Database.ClearAsync().ConfigureAwait(false);
+
+            await OrganisationEntityBuilder.Create()
+                .WithId(_org1Id)
+                .Build()
+                .InsertAsync()
+                .ConfigureAwait(false);
+
+            TestContext testContext = new TestContext();
+            _supplierRepository = testContext.SupplierRepository;
+        }
+
+        [Test]
+        public async Task ShouldGetSupplier()
+        {
+            var expected = await InsertSupplier().ConfigureAwait(false);
+
+            var result = await _supplierRepository.GetSupplierBySolutionIdAsync(_solutionId, new CancellationToken())
+                .ConfigureAwait(false);
+
+            result.SolutionId.Should().Be(_solutionId);
+            result.Description.Should().Be(_description);
+            result.Link.Should().Be(_link);
+        }
+
+        [Test]
+        public async Task ThrowNotFoundOnSupplierIfNoSupplierOrSolution()
+        {
+            var result = (await _supplierRepository.GetSupplierBySolutionIdAsync(_solutionId, new CancellationToken())
+                .ConfigureAwait(false));
+
+            result.Should().BeNull();
+        }
+
+        [Test]
+        public async Task ShouldUpdateSupplier()
+        {
+            await InsertSupplier().ConfigureAwait(false);
+
+            var supplierRequest = Mock.Of<IUpdateSupplierRequest>(s =>
+                s.SolutionId == _solutionId && s.Description == _newDescription && s.Link == _newLink);
+
+            await _supplierRepository.UpdateSupplierAsync(supplierRequest, new CancellationToken()).ConfigureAwait(false);
+
+            var supplier = await _supplierRepository.GetSupplierBySolutionIdAsync(_solutionId, new CancellationToken())
+                .ConfigureAwait(false);
+            supplier.SolutionId.Should().Be(_solutionId);
+            supplier.Description.Should().Be(_newDescription);
+            supplier.Link.Should().Be(_newLink);
+        }
+
+        [Test]
+        public async Task ShouldThrowOnUpdateIfRequestIsNull()
+        {
+            await InsertSupplier().ConfigureAwait(false);
+
+            Assert.ThrowsAsync<ArgumentNullException>(() =>
+                _supplierRepository.UpdateSupplierAsync(null, new CancellationToken()));
+        }
+
+        private async Task<SupplierEntity> InsertSupplier()
+        {
+            var expected = SupplierEntityBuilder.Create()
+                .WithOrganisation(_org1Id)
+                .WithId(_supplierId)
+                .WithSummary(_description)
+                .WithSupplierUrl(_link)
+                .Build();
+
+            await expected.InsertAsync().ConfigureAwait(false);
```

Comment:
Could this be merged with the code above (just like the Solution Entity Builder below)